.github/workflows/tests_hs-extension.yml

```
@@ -0,0 +1,80 @@
+name: Integration tests
+
+on:
+  workflow_dispatch:
+    workflow: "*"
+  pull_request_review:
+    types: [submitted, edited]
+  pull_request:
+    types: [labeled, ready_for_review, reopened]
+
+jobs:
+  integration_test:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'run-extension-tests') || github.event_name == 'workflow_dispatch' }}
+    name: Extension_${{ matrix.EXTENSION_VERSION }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        EXTENSION_VERSION: ['release', 'dev']
+
+    env:
+      MPLBACKEND: agg
+      EXTENSION: atomap etspy exspy holospy kikuchipy lumispy pyxem
+      TEST_DEPS: pytest pytest-xdist pytest-rerunfailures pytest-instafail
```

From https://github.com/hyperspy/rosettasciio/blob/c53293a1208e58575c2820d135adb779dd9b3536/pyproject.toml#L102-L109

```suggestion
      TEST_DEPS: pytest pytest-xdist pytest-rerunfailures pytest-instafail filelock pooch
```